module/Database/src/Controller/ProspectiveMemberController.php

```diff
@@ -64,6 +72,11 @@ public function finalizeAction(): Response
 
         if ($this->getRequest()->isPost()) {
             $prospectiveMember = $this->memberService->getProspectiveMember($lidnr)['member'];
+
+            if (null === $prospectiveMember) {
+                return $this->redirect()->toRoute('prospective-member');
+            }
```

Comment:
Add check here or in `finalizeSubscription` (`Member` service) to make sure that `!isPending()` on the `ProspectiveMember`.

Reply:
This check has been added (just below these lines).

Reply:
Done